File: app/src/main/java/com/commit451/gitlab/model/api/UserBasic.java
package com.commit451.gitlab.model.api;

import android.net.Uri;
import android.support.annotation.Nullable;
import android.support.annotation.StringDef;

import com.squareup.moshi.Json;

import org.parceler.Parcel;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

@Parcel
public class UserBasic extends UserSafe {

    public static final String STATE_ACTIVE = "active";
    public static final String STATE_BLOCKED = "blocked";

    @StringDef({STATE_ACTIVE, STATE_BLOCKED})
    @Retention(RetentionPolicy.SOURCE)
    public @interface State {}

    @Json(name = "id")
    long id;
    @Json(name = "state")
    @State
    String state;
    @Json(name = "avatar_url")
    String avatarUrl;
    @Json(name = "web_url")
    String webUrl;

    public UserBasic() {}

    public long getId() {
        return id;
    }

    @State
    public String getState() {
        return state;
    }

    public String getAvatarUrl() {
        return avatarUrl;
    }

    public String getWebUrl() {
        return webUrl;
    }

    @Nullable
    public Uri getFeedUrl() {
        if (webUrl == null) {
            return null;
        }
        return Uri.parse(webUrl + ".atom");
    }

    @Override
    public boolean equals(Object o) {
        if (!(o instanceof UserBasic)) {
            return false;
        }

        UserBasic user = (UserBasic) o;
        return id == user.id;
    }

    @Override
    public int hashCode() {
        return (int) (id ^ (id >>> 32));
    }
}

File: app/src/main/java/com/commit451/gitlab/model/api/UserBasic.kt
package com.commit451.gitlab.model.api

import android.support.annotation.StringDef
import com.squareup.moshi.Json
import org.parceler.Parcel


@Parcel
open class UserBasic : UserSafe() {

    companion object {

        const val STATE_ACTIVE = "active"
        const val STATE_BLOCKED = "blocked"
    }

    @StringDef(STATE_ACTIVE, STATE_BLOCKED)
    @Retention(AnnotationRetention.SOURCE)
    annotation class State

    @field:Json(name = "id")
    var id: Long = 0
    @field:Json(name = "state")
    @State
    @get:State
    var state: String? = null
    @field:Json(name = "avatar_url")
    var avatarUrl: String? = null
    @field:Json(name = "web_url")
    lateinit var webUrl: String
}
